### PR TITLE
Schema diagram

### PR DESCRIPTION
**Pull request by kwesiObuobi**
## In this PR, we did the following:
- Created a file named `schema_based_on_diagram.sql`
- Implemented a database and its tables from the given diagram
- Joined tables using many-to-many relationships
- Added FK indexes

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	CONSTRAINT fk_invoice FOREIGN KEY (invoice_id) REFERENCES invoices(id),
+	CONSTRAINT fk_treatment FOREIGN KEY (treatment_id) REFERENCES treatments(id),
+);
+
+
+
+CREATE INDEX idx_medical_histories_patient_id ON medical_histories(patient_id);
+CREATE INDEX idx_invoices_medical_history_id ON invoices(medical_history_id);
+CREATE INDEX idx_invoice_items_invoice_id ON invoice_items(invoice_id);
+CREATE INDEX idx_invoice_items_treatment_id ON invoice_items(treatment_id);
```
Comment (brytebee) [inline schema_based_on_diagram.sql:58]: Two more indexes are required.

There is a relationship between medical history and treatment which is represented by `medical_history_has_treatment`

```
CREATE INDEX ON medical_histories_has_treatments (medical_history_id);
CREATE INDEX ON medical_histories_has_treatments (treatment_id);
```

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	patient_id INT,
+	PRIMARY KEY(id),
+	CONSTRAINT fk_patients FOREIGN KEY(patient_id) REFERENCES patients(id)
+);
+
+CREATE TABLE treatments (
+	id INT GENERATED ALWAYS AS IDENTITY,
+	type VARCHAR(100),
+	name VARCHAR(100),
+	PRIMARY KEY (id)
```
Comment (brytebee) [inline schema_based_on_diagram.sql:21]: image

The relationship between the `medical_history` and `treatment` tables is many to many. All many-to-many relationships are required to be represented by a join table, `medical_history_has_treatment`

image

```sql
CREATE TABLE medical_histories_has_treatments (
    medical_history_id int refrences medical_histories(id),
    treatment_id int refrences treatments(id),
    );
```

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @kwesiObuobi & @naji-97 👋,

##

### Highlights💡
- Great practice with FK and indexes.✔️
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great DB design.✔️

##

Good job so far coming so far with the progress you've made! 💪💪
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql (hunk 1 above)
Comment (kwesiObuobi) [reply schema_based_on_diagram.sql:58]: Thanks for flagging this, @brytebee

File: schema_based_on_diagram.sql (hunk 2 above)
Comment (kwesiObuobi) [reply schema_based_on_diagram.sql:21]: Big thanks again.

**Review by Felix45 — APPROVED** [review]
# Approved :green_circle: 
Hi @naji-97 & @kwesiObuobi , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :heavy_check_mark: 
- [x] Good job setting up indexes :white_check_mark: 
- [x] Good job setting up join tables :white_check_mark: 
- [x] Implemented all the changes from the previous reviewer :white_check_mark:   

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**